src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJob.java

```diff
@@ -34,23 +35,27 @@ public String formatDollars(double amount) {
 
     @Override
     public void accept(JobContext ctx) throws Exception {
-        ctx.log("Starting to milk the cows");
+        
+            ctx.log("Starting to milk the cows");
 
-        Iterable<Commons> allCommons = commonsRepository.findAll();
+            Iterable<Commons> allCommons = commonsRepository.findAll();
 
-        for (Commons commons : allCommons) {
-            String name = commons.getName();
-            double milkPrice = commons.getMilkPrice();
-            ctx.log("Milking cows for Commons: " + name + ", Milk Price: " + formatDollars(milkPrice));
+            for (Commons commons : allCommons) {
+                String name = commons.getName();
+                double milkPrice = commons.getMilkPrice();
+                ctx.log("Milking cows for Commons: " + name + ", Milk Price: " + formatDollars(milkPrice));
 
-            Iterable<UserCommons> allUserCommons = userCommonsRepository.findByCommonsId(commons.getId());
+                Iterable<UserCommons> allUserCommons = userCommonsRepository.findByCommonsId(commons.getId());
 
-            for (UserCommons userCommons : allUserCommons) {
-                milkCows(ctx, commons, userCommons);
-            }
-        }
-
-        ctx.log("Cows have been milked!");
+                for (UserCommons userCommons : allUserCommons) {
+                    if(userCommons.gameInProgress()){
```

Comment:
Why are we checking this at the userCommons level?

The game is in progress, or not, at the Commons level, not at the UserCommons level.

It doesn't make sense to iterate over every player and make the same check for each one. This check should be at the outer loop.